src/TwigEnvironmentFactory.php

```diff
@@ -102,7 +103,7 @@ public function __invoke(ContainerInterface $container) : Environment
             'cache'            => $debug ? false : $cacheDir,
             'debug'            => $debug,
             'strict_variables' => $debug,
-            'auto_reload'      => $debug,
+            'auto_reload'      => $config['auto_reload'] ?? $debug,
```

Comment:
I think I prefer `$debug ?? $config['auto_reload']`. This would always reload in debug mode and otherwise if `auto_reload` in production mode.

Reply:
I would say it is bad idea. For example, we have config `['debug' => false, 'auto_reload' => true]`, so `auto_reload` will be always false. From Twig docs `auto_reload` option must override debug value, if this option does not exist, it gets value from debug option.

Reply:
@xtreamwayz , can I resolve discussion?